packages/docusaurus-theme-classic/src/theme/BlogPostItem/index.js

```diff
@@ -92,13 +93,13 @@ function BlogPostItem(props) {
   };
 
   return (
-    <div>
+    <article {...attrs}>
```

Comment:
No, we don't want to channel props through props, it makes it harder to know what props are being used and which aren't. Accepting `className` props is also an anti-pattern that we don't want to encourage. If `article` defines its own `className` in future, it will work when the parents don't pass in a `className` but will break when the parents do. This kind of bugs are hard to detect and reliably prevent, so we don't want to allow that at all.

Reply:
>  Accepting className props is also an anti-pattern that we don't want to encourage. 

And what should I do then?

Reply:
>  If article defines its own className in future, it will work when the parents don't pass in a className but will break when the parents do.

We can merge classes with `classnames` lib, right? Why is that bad? I agree with passing rest props, but why can't extract `className` if all CSS classes are merged later?

Reply:
Used a different approach, I hope it will be good for you https://github.com/facebook/docusaurus/pull/2003/commits/83eeafc0de1318e260d72592de37c7be0722f133

Reply:
> We can merge classes with classnames lib, right? Why is that bad?

Which class get priority? The component's classes or the user's classes? Either approach is bad as it leads to unpredictable behavior. In this scenario:

```
.user-class { padding: 0; }
.component-class { padding: 12px; }
```

Both classes affect `padding` and again this makes the final result dependent on CSS ordering. Imagine the user added `background-color: red` to `.user-class` and in a future release we add `background-color: orange` to `.component-class`, it could break the user's CSS. Also, since we use webpack with code splitting, the CSS loading order is hard to reason about and can lead to obscure bugs.

The only way is to not allow any merging of CSS classes. This is our practice for Facebook components. For common properties that people like to override, we allow passing in of `margin`/`padding` props. What you've done here is acceptable as we're still in control.

Reply:
Thanks for the clarification, in this case, the situation is somewhat different, since we are dealing with an _internal_ component, I mean, an end user can set own CSS class only if component will be swizzled, initially I did not take this fact into account.